File: 1460-make-two-arrays-equal-by-reversing-subarrays/1460-make-two-arrays-equal-by-reversing-subarrays.java
class Solution {
    public boolean canBeEqual(int[] target, int[] arr) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int n = target.length;
        for(int i=0;i<n;i++) {
            map.put(target[i], map.getOrDefault(target[i],0)+1);
        }
        for(int i=0;i<n;i++) {
            if(map.containsKey(arr[i])) {
                map.put(arr[i], map.get(arr[i]) - 1);
                if(map.get(arr[i]) == 0) {
                    map.remove(arr[i]);
                }
            }
        }
        if(!map.isEmpty()) {
            return false;
        }
        return true;
    }
}

File: 1460-make-two-arrays-equal-by-reversing-subarrays/1460-make-two-arrays-equal-by-reversing-subarrays.cpp
class Solution {
public:
    bool canBeEqual(vector<int>& target, vector<int>& arr) {
        unordered_map<int, int> mpp;
        int n = target.size();
        for(int i=0;i<n;i++) {
            mpp[target[i]]++;
        }
        for(int i=0;i<n;i++) {
            if(mpp.find(arr[i])!= mpp.end()) {
                mpp[arr[i]]--;
                if(mpp[arr[i]] == 0) {
                    mpp.erase(arr[i]);
                }
            }
        }
        if(!mpp.empty()) {
            return false;
        }
        return true;
    }
};